util-linux: add include linux/unistd.h patch

namiltd opened the pull request:
If SYS_statmount/SYS_listmount is not provided by the libc, util-linux will fall back to __NR_statmount/__NR_listmount from the kernel UAPI headers.
However it is not guaranteed that these symbols are actually visible in mount-api-utils.

Include linux/unistd.h which provides syscall numbers. While this header is specific to Linux, the code is already using linux/mount.h.

t-8ch commented:
I think this should go to `packages/utils/util-linux` and not `tools/util-linux`.

namiltd committed:
util-linux: add include linux/unistd.h patch

If SYS_statmount/SYS_listmount is not provided by the libc,
util-linux will fall back to __NR_statmount/__NR_listmount from the
kernel UAPI headers.
However it is not guaranteed that these symbols are actually visible in
mount-api-utils.

Include linux/unistd.h which provides syscall numbers.
While this header is specific to Linux, the code is already using
linux/mount.h.

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Link: https://github.com/openwrt/openwrt/pull/18539
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

namiltd commented:
That's right, fixed.

httpstorm commented:
@namiltd Thanks!
This fixes the build error on macOS 15.4.1 Intel host, when using kernel 6.12.

Build log before: [2025-04-19 util-linux 6.12.txt](https://httpstorm.com/share/.openwrt/test/2025-04-18_linux/2025-04-19%20util-linux%206.12.txt)
Build log after: [2025-04-20 util-linux 6.12.txt](https://httpstorm.com/share/.openwrt/test/2025-04-18_linux/2025-04-20%20util-linux%206.12.txt)

The next package which fails with kernel 6.12 is `xtables-addons`: [2025-04-20 xtables-addons 6.12.txt](https://httpstorm.com/share/.openwrt/test/2025-04-18_linux/2025-04-20%20xtables-addons%206.12.txt)

namiltd commented:
This requires creating an appropriate patch on https://github.com/openwrt/packages

namiltd commented:
PR created: https://github.com/openwrt/packages/pull/26353

namiltd commented:
PR https://github.com/openwrt/packages/pull/26353 accepted
